support/package_graphicsmagick

```diff
@@ -1,120 +1,93 @@
+pushd $(dirname $0)/.. >>/dev/null
+home=$(pwd)
+popd >>/dev/null
 
-echo ">>> Uploading libpng artifact"
-echo "    to ${S3_BUCKET}/${LIBPNG_REMOTE_PATH}"
-echo "    from ${libpng_local_path}"
+arr=($(parse_version "$GRAPHICS_MAGICK_VERSION"))
+major=${arr[0]}
+minor=${arr[1]}
+patch=${arr[2]}
+VENDOR_GRAPHICS_MAGICK_ARCHIVE_URL="${VENDOR_GRAPHICS_MAGICK_ARCHIVE_URL:-"https://hg.osdn.net/view/graphicsmagick/GM/archive/GraphicsMagick-${major}_${minor}_${patch}.tar.gz"}"
 
-cd /app/vendor/libpng && tar -czvf $libpng_local_path .
-$basedir/aws/s3 put $S3_BUCKET $STACK/$LIBPNG_REMOTE_PATH $libpng_local_path
-
-echo ">>> Building nasm"
-
-nasm_local_path=$tempdir/nasm-${NASM_VERSION}.tgz
-
-# build and package nasm for heroku
-cp -R $home/vendor/nasm-${NASM_VERSION} $tempdir
-cd $tempdir/nasm-${NASM_VERSION}
-./configure --prefix /app/vendor/nasm && make install
-
-echo ">>> Uploading nasm artifact"
-echo "    to ${S3_BUCKET}/${NASM_REMOTE_PATH}"
-echo "    from ${nasm_local_path}"
-
-cd /app/vendor/nasm && tar -czvf $nasm_local_path .
-$basedir/aws/s3 put $S3_BUCKET $STACK/$NASM_REMOTE_PATH $nasm_local_path
-
-echo ">>> Building libjpeg-turbo"
-
-libjpeg_turbo_local_path=$tempdir/libjpeg_turbo-${LIBJPEG_TURBO_VERSION}.tgz
-
-cp -R $home/vendor/libjpeg-turbo-${LIBJPEG_TURBO_VERSION} $tempdir
-cd $tempdir/libjpeg-turbo-${LIBJPEG_TURBO_VERSION}
-PATH=$PATH:/app/vendor/nasm/bin bash -c './configure --prefix /app/vendor/libjpeg_turbo && make install'
-
-echo ">>> Uploading libjpeg-turbo artifact"
-echo "    to ${S3_BUCKET}/${LIBJPEG_TURBO_REMOTE_PATH}"
-echo "    from ${libjpeg_turbo_local_path}"
-
-cd /app/vendor/libjpeg_turbo && tar -czvf $libjpeg_turbo_local_path .
-$basedir/aws/s3 put $S3_BUCKET $STACK/$LIBJPEG_TURBO_REMOTE_PATH $libjpeg_turbo_local_path
+basedir="$(cd -P "$(dirname "$0")" && pwd)"
 
 echo ">>> Downloading graphicsmagick"
 
-gm_local_path=$tempdir/graphcismagick-${GRAPHICS_MAGICK_VERSION}.tgz
-gm_sources=$tempdir/graphicsmagick-${GRAPHICS_MAGICK_VERSION}.tar.gz
-#curl ftp://ftp.graphicsmagick.org/pub/GraphicsMagick/1.3/GraphicsMagick-${GRAPHICS_MAGICK_VERSION}.tar.gz > $gm_sources
-curl http://78.108.103.11/MIRROR/ftp/GraphicsMagick/1.3/GraphicsMagick-${GRAPHICS_MAGICK_VERSION}.tar.gz > $gm_sources
-
-tar -C $tempdir -x -z -f $gm_sources
+# vendor directories
+vendored_graphicsmagick_dir="$(mktmpdir graphicsmagick)"
+archive_download "$VENDOR_GRAPHICS_MAGICK_ARCHIVE_URL" "$vendored_graphicsmagick_dir"
 
 echo ">>> Building graphicsmagick"
 
-# build and package imagemagick for heroku
-cd $tempdir/GraphicsMagick-${GRAPHICS_MAGICK_VERSION}
-./configure --prefix /app/vendor/graphicsmagick && make install
+gm_local_path=/tmp/graphicsmagick-${GRAPHICS_MAGICK_VERSION}.tar.gz
+gm_lib_path=${BUILD_DIR}/vendor/graphicsmagick
+mkdir -p $gm_lib_path
+
+pushd $vendored_graphicsmagick_dir
+./configure
+make
+make DESTDIR=$gm_lib_path install
+popd
+
+echo ">>> Creating graphics magick archive"
```

Comment:
nitpicky: this is really me being picky but I don't like the fact that GraphicsMagick is spelled with various different spellings in the same file that bother me. I tihnk it should be `GraphicsMagick` everywhere